test_regress/t/t_vpi_dump.cpp

```diff
@@ -0,0 +1,204 @@
+        // vpiSpecParam,         // Don't care
+        // vpiParamAssign,       // Aldec SEGV on mixed language
+        // vpiDefParam,          // Don't care
+        vpiPrimitive, vpiPrimitiveArray,
+        // vpiContAssign,        // Don't care
+        vpiProcess,  // Don't care
+        vpiModPath, vpiTchk, vpiAttribute, vpiPort, vpiInternalScope,
+        // vpiInterface,         // Aldec SEGV on mixed language
+        // vpiInterfaceArray,    // Aldec SEGV on mixed language
+    };
+
+    // append base class vpiInstance members
+    module_options.insert(module_options.begin(), instance_options.begin(),
+                          instance_options.end());
+
+    std::vector<int32_t> struct_options = {
+        vpiNet,       vpiReg,       vpiRegArray,       vpiMemory,
+        vpiParameter, vpiPrimitive, vpiPrimitiveArray, vpiAttribute,
+        vpiMember,
+    };
+
+    return decltype(iterate_over){
+        {vpiModule, module_options},
+        {vpiInterface, instance_options},
+        {vpiGenScope, module_options},
+
+        {vpiStructVar, struct_options},
+        {vpiStructNet, struct_options},
+
+        {vpiNet,
+         {
+             // vpiContAssign,        // Driver and load handled separately
+             // vpiPrimTerm,
+             // vpiPathTerm,
+             // vpiTchkTerm,
+             // vpiDriver,
+             // vpiLocalDriver,
+             // vpiLoad,
+             // vpiLocalLoad,
+             vpiNetBit,
+         }},
+        {vpiNetArray,
+         {
+             vpiNet,
+         }},
+        {vpiRegArray,
+         {
+             vpiReg,
+         }},
+        {vpiMemory,
+         {
+             vpiMemoryWord,
+         }},
+        {vpiPort,
+         {
+             vpiPortBit,
+         }},
+        {vpiGate,
+         {
+             vpiPrimTerm,
+             vpiTableEntry,
+             vpiUdpDefn,
+         }},
+        {vpiPackage,
+         {
+             vpiParameter,
+         }},
+    };
+}();
+
+void modDump(vpiHandle it, int n) {
+
+    if (n > 8) {
+        printf("going too deep\n");
+        return;
+    }
+
+    while (TestVpiHandle hndl = vpi_scan(it)) {
+        for (int i = 0; i < n; i++) printf("    ");
+        auto type = vpi_get(vpiType, hndl);
+        auto name = vpi_get_str(vpiName, hndl);
+        auto fullname = vpi_get_str(vpiFullName, hndl);
+        printf("%s (%s) %s\n", name, strFromVpiObjType(type), fullname);
+        // vpi_get(vpiSize, hndl),
+        if (type == vpiParameter || type == vpiConstType) {
+            for (int i = 0; i < n + 1; i++) printf("    ");
+            printf("vpiConstType=%s\n", strFromVpiConstType(vpi_get(vpiConstType, hndl)));
+        }
+
+        if (iterate_over.find(type) == iterate_over.end()) { continue; }
+        for (int type : iterate_over.at(type)) {
+            auto subIt = vpi_iterate(type, hndl);
+            if (subIt) modDump(subIt, n + 1);
+        }
+    }
+}
+
+PLI_INT32 vpi_main(t_cb_data* data) {
```

Comment:
Please don't name any test functions with vpi_* to avoid confusion with true vpi library routines.  Use e.g. test_*